client/state/data-layer/wpcom/sites/jitm/index.js

```diff
@@ -39,7 +39,11 @@ const process = {
 /**
  * Routes in which to specifically prevent from displaying JITMs.
  */
-const routesToIgnore = [ '/jetpack/connect/site-type', '/jetpack/connect/site-topic' ];
+const routesToIgnore = [
+	'/jetpack/connect/user-type',
```

Comment:
Nit: I'd put this at the bottom to preserve the order of steps, but I'll leave that to your judgement.